esp32/door-sensor/src/main.cpp

```diff
@@ -0,0 +1,89 @@
+#include <Arduino.h>
+#include <HTTPClient.h>
+#include <WiFi.h>
+
+#include "config.h"
+
+#define DOOR_SENSOR_PIN 15
+#define DEBOUNCE_DELAY 500
+
+int current_state;
+int last_state;
+unsigned long last_debounce_time = 0;
+
+void sendDoorState(int state);
+
+void setup() {
+  // Initialize serial communication for debugging
+  Serial.begin(115200);
+  while (!Serial)
+    delay(100);
+  Serial.println("\n");
+
+  // Connect to WiFi
+  Serial.println("Connecting to WiFi...");
+  WiFi.begin(WIFI_SSID, WIFI_PASSWORD);
+  while (WiFi.status() != WL_CONNECTED)
+    delay(100);
+  Serial.println("Connected: " + String(WIFI_SSID));
+
+  // Set up the door sensor pin as an input with pull-up resistor
+  pinMode(DOOR_SENSOR_PIN, INPUT_PULLUP);
+
+  // Initial reading of the door sensor
+  current_state = digitalRead(DOOR_SENSOR_PIN);
+  Serial.println(current_state == HIGH ? "\nDoor opened" : "\nDoor closed");
+  sendDoorState(current_state);
+}
+
+void loop() {
+  int raw_state = digitalRead(DOOR_SENSOR_PIN);
+
+  // If the raw state changes, reset the debounce timer
+  if (raw_state != last_state)
+    last_debounce_time = millis();
+
+  // Check if the debounce delay has passed since the last change
+  if ((millis() - last_debounce_time) > DEBOUNCE_DELAY) {
+    // Check if the raw state differs from the current state
+    if (raw_state != current_state) {
+      // Update the current state and print the door status
+      current_state = raw_state;
+      Serial.println(current_state == HIGH ? "\nDoor opened" : "\nDoor closed");
+
+      // Send the door state to the server
+      sendDoorState(current_state);
+    }
+  }
+
+  // Update the last state for the next iteration
+  last_state = raw_state;

```

Comment:
Would it be worth it so just send a heartbeat of the current door status? Then if the ESP32 crashes, the bot can send an "unavailable" message? as if the bot hadn't heard from the esp32 within 20 minutes or what have you, the bot could consider the ESP32 as disconnected.

Reply:
https://github.com/GryphonRacingFSAE/Discord-Bot/pull/20/commits/eef9ffd10c4e1efb937883f5f0b3b33e5a08300d